packages/gestalt-datepicker/src/DatePickerTextField.js

```diff
@@ -0,0 +1,100 @@
+// @flow strict-local
+import * as React from 'react';
+import { Box, Icon, Touchable, TextField } from 'gestalt';
+import PropTypes from 'prop-types';
+
+type OwnProps = {|
+  accessibilityLabelIcon: string,
+  isRTL?: boolean,
+|};
+
+// InjectedProps are props that Datepicker adds on to DatePickerTextField.
+// Datepicker takes this props and then funnels them to DatePickerTextField.
+type InjectedProps = {|
+  disabled?: boolean,
+  id?: string,
+  onClick?: () => void,
+  placeholder?: string,
+  value?: string,
+|};
+
+type Props = {|
+  ...OwnProps,
+  ...InjectedProps,
+  forwardedRef?: React.Ref<'input'>,
+|};
+
+function DatePickerTextField(props: Props) {
+  const {
+    accessibilityLabelIcon,
+    disabled,
+    forwardedRef,
+    id,
+    isRTL = false,
+    onClick,
+    placeholder,
+    value,
+  } = props;
+
+  return (
+    <Touchable disabled={disabled} onTouch={onClick}>
```

Comment:
@AlbertCarreras can we avoid wrapping this in a `<Touchable />` - otherwise we will have 2 focus states, one on Touchable and one on TextField:

![image](https://user-images.githubusercontent.com/127199/84450198-4e327e00-ac04-11ea-88a1-d17fe62369af.png)
![image](https://user-images.githubusercontent.com/127199/84450209-55598c00-ac04-11ea-9ba2-4a4d8bde20aa.png)

Reply:
fixed, comment here https://www.loom.com/share/e4a93927ab0d46f4be67689b8d70becf